src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1345 @@ static ColliderCylinderInit D_80B37A5C = {
+    this->actionTimer = 20;
+    this->isInvisible = false;
+    this->action = WOLFOS_ACTION_WAIT_TO_APPEAR;
+    this->actor.flags &= ~1;
+    this->actor.scale.y = 0.0f;
+    this->actor.gravity = 0.0f;
+    EnWf_SetupAction(this, EnWf_WaitToAppear);
+}
+
+void EnWf_WaitToAppear(EnWf* this, GlobalContext* globalCtx) {
+    if (this->actionTimer >= 6) {
+        this->actor.world.pos.y = this->actor.home.pos.y - 5.0f;
+
+        if (this->actor.xzDistToPlayer < 240.0f) {
+            this->actionTimer = 5;
+            this->actor.flags |= 1;
+
+            if ((this->actor.params != 0) && (this->switchFlag != 0xFF)) {
+                func_800F5ACC(0x38); // Mini-Boss Battle Theme
+            }
+        }
+    } else if (this->actionTimer != 0) {
+        this->actor.scale.y += (this->actor.scale.x * 0.2f);
+        this->actor.world.pos.y += 0.5f;
+        Math_SmoothStepToF(&this->actor.shape.shadowScale, 70.0f, 1.0f, 14.0f, 0.0f);
+        this->actionTimer--;
+
+        if (this->actionTimer == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_WOLFOS_APPEAR);
+        }
+    } else if (SkelAnime_Update(&this->skelAnime)) {
+        this->actor.scale.y = this->actor.scale.x;
+        this->actor.gravity = -2.0f;
+        EnWf_SetupWait(this);
+    }
+}
+
+void EnWf_SetupWait(EnWf* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gWolfosWaitingAnim, -4.0f);
+    this->action = WOLFOS_ACTION_WAIT;
+    this->actionTimer = (Rand_ZeroOne() * 10.0f) + 2.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnWf_SetupAction(this, EnWf_Wait);
+}
+
+void EnWf_Wait(EnWf* this, GlobalContext* globalCtx) {
+    Player* player;
+    s32 pad;
+    s16 angle;
+
+    player = GET_PLAYER(globalCtx);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->headTilt != 0) {
+        angle = (this->actor.yawTowardsPlayer - this->actor.shape.rot.y) - this->headRot.y;
+
+        if (ABS(angle) > 0x2000) {
+            this->headTilt--;
+            return;
+        }
+
+        this->headTilt = 0;
+    }
+
+    angle = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    angle = ABS(angle);
+
+    if (!(EnWf_DodgeRanged(globalCtx, this))) {
+        // Only use of unk_2E0: never not zero, so this if block never runs
+        if (this->unk_2E0 != 0) {
+            this->unk_2E0--;
+
+            if (angle >= 0x1FFE) {
+                return;
+            }
+            this->unk_2E0 = 0;
+        } else if (EnWf_ChangeAction(globalCtx, this, 0)) {
+            return;
+        }
+
+        angle = player->actor.shape.rot.y - this->actor.shape.rot.y;
+        angle = ABS(angle);
+
+        if ((this->actor.xzDistToPlayer < 80.0f) && (player->swordState != 0) && (angle >= 0x1F40)) {
+            this->actor.shape.rot.y = this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+            EnWf_SetupRunAroundPlayer(this);
+        } else {
+
+            this->actionTimer--;
```

Comment:
i thought this review comment went through before but it apparently didnt?
anyway, missed this one last time, my bad

Reply:
I remember seeing it 🤔